### DISCUSSION TITLE
http post:  how to set values

### DISCUSSION THREAD

**Discussion by xal88** [Lenti84/VdMot_Controller#90] [Show and tell]
Hi all, 
I can set a single value via the http- json like:

_# requests.post("http://172.21.14.153/protconfig", json = {"pubTarget":0})_

But how can I set e.g. values if the json is a list?
e.g. /tempsconfig[0]['offset']
or  /valves[1]['targetpos]

Passing the whole JSON with all list elements gives error 400.

Can someone give an example, eg. for **tTarget** or **tValue**?
Kind regards
Xal

**Comment by harvey637** [comment]
Hi Gargano, hope you are safe back!
@xal88: suprise with the /setvalve command 

As summer gets hotter can you please explain a bit on the "heating/cooling" configuration?

I remember something like
`curl -s -H 'Content-Type: application/json' -X POST -d "{ \"valve\":$valve , \"ctrlValue\":$new_temp , \"ctrlTarget\":$set_temp }" http://192.168.xxx.xxx/setvalve`
for the setting of a single valve ("valve" = index of valve 1-14) to measured temperature ("ctrlValue" = room temperature in °C) and the  requested target temperature ("ctrlTarget" = open/close valve to reach this target temperature in °C)

As I remember I have to add something like "heatControl":1" for heating and "heatControl":2" for cooling?
Can I set this for ALL valves in one POST, as it does not make much sense heating one room and cooling another  :-)

So it makes 
`curl -s -H 'Content-Type: application/json' -X POST -d "{ \"heatControl\":2 }" http://192.168.xxx.xxx/setvalve`
to set all valves to cooling, and
`curl -s -H 'Content-Type: application/json' -X POST -d "{ \"heatControl\":1 }" http://192.168.xxx.xxx/setvalve`
to set all valves to heating (which is the default)?

I think this also helps @xal88 with his question, as this /setvalve is not yet documented. Hopefully it (this command and the heating/cooling) is also merged into the basic lenti version soon.

cu Harvey
PS: lets continue some talk on "mysterious/unneccessary valve opening after set-temperature jump" if you have some spare time for this - thank  you 
Harvey

**Comment by SurfGargano** [comment]
For writing json valves control pls see https://github.com/Lenti84/VdMot_Controller/wiki/JSON-Interface

**Reply by xal88** [reply]
thanks for this additional documentation.
Could you please add an example of how to change control for [heating|cooling|manual]?

The [setvalve example from harvey637 ](https://github.com/Lenti84/VdMot_Controller/discussions/90#discussioncomment-6302796) was not working for me.
kind regards
Alf

**Reply by SurfGargano** [reply]
I have updated the wiki.
I use the postman to check the json interface.

The setvalve example from harvey is wrong.  The url is for setvalve while the payload is for valvesctrlconfig

**Comment by harvey637** [comment]
![grafik](https://github.com/Lenti84/VdMot_Controller/assets/10156426/e2c0c308-fd59-40da-9aad-0abf243b0a8b)

just my imagination of the status display with the heating/cooling mode for each valve.
valve 8 in this fictious example is NOT allowed to cool, only to heat :-), so no cooling mode displayed for this valve.
Temperatures are just live examples and do NOT reflect any correct temperatures for cooling!

please explain a bit about the setting of the cooling/heating mode, the ALLOW heat/cool I guess I do understand, but not the
change of the algorithm (tValue < tTarget versus tValue > tTarget ...).

looking forward to hear more from you, thank you!
cu
Harvey

**Reply by SurfGargano** [reply]
The explaination is done in the control section here https://github.com/Lenti84/VdMot_Controller/wiki/Web-UI

On/Heating : The pi-controller is on, a positive difference of target-values opens the valves
On/Cooling : The pi-controller is on, a negative difference of target-values opens the valves
